Ghidra/Processors/Xtensa/data/languages/xtensaInstructions.sinc

```diff
@@ -0,0 +1,1358 @@
+        goto inst_next;
+    ar = as;
+}
+
+# MOVLTZ.S - Move Single if Less Than Zero, pg. 424.
+:movltz.s fr, fs, at is op2 = 0b1010 & op1 = 0b1011 & fr & fs & at & op0 = 0 {
+    if (at s>= 0)
+        goto inst_next;
+    fr = fs;
+}
+
+# MOVNEZ - Move if Not Equal to Zero, pg. 425.
+:movnez ar, as, at is op2 = 0b1001 & op1 = 0b0011 & ar & as & at & op0 = 0 {
+    if (at == 0)
+        goto inst_next;
+    ar = as;
+}
+
+# MOVNEZ.S - Move Single if Not Equal to Zero, pg. 426.
+:movnez.s fr, fs, at is op2 = 0b1001 & op1 = 0b1011 & fr & fs & at & op0 = 0 {
+    if (at == 0)
+        goto inst_next;
+    fr = fs;
+}
+
+# MOVT - Move if True, pg. 428.
+:movt ar, as, bt is op2 = 0b1101 & op1 = 0b0011 & ar & as & bt & op0 = 0 {
+    if (!bt)
+        goto inst_next;
+    ar = as;
+}
+
+# MOVT.S - Move Single if True, pg. 429.
+:movt.s fr, fs, bt is op2 = 0b1101 & op1 = 0b1011 & fr & fs & bt & op0 = 0 {
+    if (!bt)
+        goto inst_next;
+    fr = fs;
+}
+
+# MSUB.S - Multiply and Subtract Single, pg. 430.
+:msub.s fr, fs, ft is op2 = 0b0101 & op1 = 0b1010 & fr & fs & ft & op0 = 0 {
+    fr = fr f- (fs f* ft);
+}
+
+# MUL.S - Multiply Single, pg. 435.
+:mul.s fr, fs, ft is op2 = 0b0010 & op1 = 0b1010 & fr & fs & ft & op0 = 0 {
+    fr = fs f* ft;
+}
+
+# MUL16S - Multiply 16-bit Signed, pg. 436.
+:mul16s ar, as, at is op2 = 0b1101 & op1 = 0b0001 & ar & as & at & op0 = 0 {
+    ar = sext(as:2) * sext(at:2);
+}
+
+# MUL16U - Multiply 16-bit Unsigned, pg. 437.
+:mul16u ar, as, at is op2 = 0b1100 & op1 = 0b0001 & ar & as & at & op0 = 0 {
+    ar = zext(as:2) * zext(at:2);
+}
+
+# MULL - Multiply Low, pg. 450.
+:mull ar, as, at is op2 = 0b1000 & op1 = 0b0010 & ar & as & at & op0 = 0 {
+    ar = as * at;
+}
+
+# MULSH - Multiply Signed High, pg. 455.
+:mulsh ar, as, at is op2 = 0b1011 & op1 = 0b0010 & ar & as & at & op0 = 0 {
+    local s64:8 = sext(as);
+    local t64:8 = sext(at);
+    local p:8 = (s64 * t64);
+    ar = p(4);
+}
+
+# MULUH - Multiply Unsigned High, pg. 456.
+:muluh ar, as, at is op2 = 0b1010 & op1 = 0b0010 & ar & as & at & op0 = 0 {
+    local s64:8 = zext(as);
+    local t64:8 = zext(at);
+    local p:8 = (s64 * t64);
+    ar = p(4);
+}
+
+# NEG - Negate, pg. 457.
+:neg ar, at is op2 = 0b0110 & op1 = 0 & ar & as = 0 & at & op0 = 0 {
+    ar = -at;
+}
+
+# NEG.S - Negate Single, pg. 458.
+:neg.s fr, fs is op2 = 0b1111 & op1 = 0b1010 & fr & fs & at = 0b0110 & op0 = 0 {
+    fr = f- fs;
+}
+
+# NOP - No Operation, pg. 459.
+:nop is op2 = 0 & op1 = 0 & ar = 0b0010 & as = 0 & at = 0b1111 & op0 = 0 { }
+
+# NOP.N - Narrow No Operation, pg. 460.
+:nop.n is n_ar = 0b1111 & n_as = 0 & n_at = 0b0011 & n_op0 = 0b1101 { }
+
+# NSA - Normalization Shift Amount, pg. 461.
+:nsa at, as is op2 = 0b0100 & op1 = 0 & ar = 0b1110 & as & at & op0 = 0 {
+    # https://stackoverflow.com/questions/54772520/normalizing-a-two-complement-number
+    at = nsa(as);
+}
+
+# NSAU - Normalization Shift Amount Unsigned, pg. 462. (Count leading zeros)
```

Comment:
`lzcount` is a new built-in pcode that might simplify this assuming it is `Count leading zeros`

Reply:
Addressed in 29bf647375fd9c5f6f0e9053e85429debfd7ec65.

This definitely cleans up the de-compiler output for the NSAU instruction. I was unaware of this new pcode.

Reply:
Also looks like you can replace the `NSA` implementation `nsa()` with `lzcount(~as)` (and delete pcodeop nsa in the other file)

Reply:
Addressed in afa0ff6aac5b9895710af51f331a010ff726e268. I saw this but wasn't sure myself whether this would be applicable. Thank you!